apps/babelstream/package.py

```diff
@@ -128,8 +128,9 @@ class Babelstream(CMakePackage,CudaPackage,ROCmPackage):
     # depends_on("raja", when="+raja")
 
     #OpenCL Dependency
-    depends_on("pocl@1.5", when="+opencl")
-    depends_on("gcc@9.2.0", when="+opencl")
+    depends_on("pocl@1.5", when="+ocl")
+    depends_on("gcc@9.2.0", when="+ocl")
+    depends_on("cmake@3.23.2", when="+ocl")
```

Comment:
We need this change because it picks up the `cmake@3.11` from the environment and it does not satisfy the conditions for this build ( needs to be at least 3.14) so either I provide for all options to use `3.23.2` on `package.py` or better option change it directly from environment if other builds won't require this specific version ? @giordano

Reply:
If you need `3.14` you can specify exactly that version, instead of the overly restrictive `3.23.2`: if a system happens to have CMake 3.14 can use the global installation, otherwise a newer CMake will be automatically compiled.

Also, I presume CMake is a build-only dependency, so you should add `type="build"`.

Reply:
Also, why is CMake a conditional dependency?  Don't you always need it?

Reply:
I added it as a dependency because the environment in [here](https://github.com/ukri-excalibur/excalibur-tests/blob/c2b80844eb7096842c595dac07926011659f72a1/spack-environments/isambard-cascadelake/compute-node/spack.yaml#L100-L103) has cmake 3.11.4 so I removed that in my own local version and added `depends_on` but as you said if we remove it from environment it will pick up the latest one so we do not need to mention the version